### PR TITLE
Make some functions work with Milvus

### PR DESCRIPTION
**Pull request by hoanq1811**
**Description**
Make some functions work with Milvus:
1. get_ids: Get primary keys by field in the metadata
2. delete: Delete one or more entities by ids
3. upsert: Update/Insert one or more entities

**Issue**
None
**Dependencies**
None
**Tag maintainer:**
@hwchase17 
**Twitter handle:**
None

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/milvus.py [hunk 1]
```diff
@@ -821,3 +821,87 @@ def from_texts(
...
+        except MilvusException as exc:
+            logger.error(
+                "Failed to get pks: %s error: %s", self.collection_name, exc
+            )
+            raise exc
+        pks = [item.get("pk") for item in query_result]
+        
+        return pks
+ 
+    def delete(self, pks: List[int] | None = None, **kwargs: Any) -> bool | None:
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/milvus.py:855]: can we use the method signature inherited from parent class?
```python
    def delete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> Optional[bool]:
```

**Comment by baskaryan** [comment]
cc @filip-halt

File: libs/langchain/langchain/vectorstores/milvus.py (hunk 1 above)
Comment (hoanq1811) [reply libs/langchain/langchain/vectorstores/milvus.py]: Yes, I’ve updated it. You can check it now

**Comment by hoanq1811** [comment]
@baskaryan @filip-halt 
Hey, my friends, 
Please help me to review,
Thanks & Best Regards

**Comment by hoanq1811** [comment]
Hi @baskaryan and @hwchase17 ,
I saw that the structure code was updated and the **delete** feature was implemented.
We have a problem: If the developers used the old version, they can't know what the primary key field (id) was
So I have updated a new feature: get_ids. That helps us easily get the primary key by expression
Meanwhile, I also added the **upsert** feature to insert or update documents

**Comment by fzliu** [comment]
@hoanq1811 Looks great! Thanks for contributing.

File: libs/community/langchain_community/vectorstores/milvus.py [hunk 2]
```diff
@@ -989,3 +989,69 @@ def _parse_document(self, data: dict) -> Document:
             page_content=data.pop(self._text_field),
             metadata=data.pop(self._metadata_field) if self._metadata_field else data,
         )
+
+    def get_ids(self,
```
Comment (fzliu) [inline libs/community/langchain_community/vectorstores/milvus.py:993]: Minor point - I would change this to `get_pks` to avoid confusion.

**Review by baskaryan — APPROVED** [review]

**Comment by yiwangdr** [comment]
ooc, is the primary purpose to support upsert? You can call the milvus upsert api directly (https://milvus.io/docs/upsert_entities.md)

File: libs/community/langchain_community/vectorstores/milvus.py (hunk 2 above)
Comment (hoanq1811) [reply libs/community/langchain_community/vectorstores/milvus.py]: Hi @fzliu ,
Yup. I totally agree. So I have changed it by moving `get_ids` to `get_pks`

**Comment by hoanq1811** [comment]
> ooc, is the primary purpose to support upsert? You can call the milvus upsert api directly (https://milvus.io/docs/upsert_entities.md) 

Hi @yiwangdr 
Thank for your suggestion.

Actually, I made 3 features: delete, get_pks, upsert before. But now in the latest version, I found that the `delete` feature has been integrated, so I updated it again
 
For the `upsert` feature, last time I saw it wasn't supported yet. But now, it's supported, so I will look into it for integration with the langchain.

**Pull request merged**